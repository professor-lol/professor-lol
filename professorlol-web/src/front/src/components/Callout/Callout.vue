<template>
  <div :class="classList">
    <slot>Callout</slot>
  </div>
</template>

<script>
export default {
  props: {
    variant: {
      type: String,
      default: ''
    }
  },
  computed: {
    classList () {
      return [
        'callout',
        this.calloutVariant
      ]
    },
    calloutVariant () {
      return this.variant ? `callout-${this.variant}` : ''
    }
  }
}
</script>
