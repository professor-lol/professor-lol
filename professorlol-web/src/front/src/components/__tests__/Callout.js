import { mount } from "@vue/test-utils";
import Callout from "../Callout/Callout";

describe("Callout.vue", () => {
  // Inspect the raw component options
  it("should have default props", () => {
    const wrapper = mount(Callout, {
      propsData: {
        variant: ''
      }
    })
    expect(wrapper.props().variant).toBe('')
  });
  it('has classList computed property', () => {
    expect(typeof Callout.computed.classList).toBe('function')
  })
  it('has calloutVariant computed property', () => {
    expect(typeof Callout.computed.calloutVariant).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(Callout)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.element.textContent).toEqual('Callout')
    expect(wrapper.classes()).toContain('callout')
  })
});
