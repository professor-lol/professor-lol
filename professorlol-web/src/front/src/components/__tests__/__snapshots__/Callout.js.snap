// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callout.vue renders correctly 1`] = `
<div
  class="callout"
>
  Callout
</div>
`;
